control/tests/nyquist_test.py

```diff
@@ -0,0 +1,292 @@
+"""nyquist_test.py - test Nyquist plots
+
+RMM, 30 Jan 2021
+
+This set of unit tests covers various Nyquist plot configurations.  Because
+much of the output from these tests are graphical, this file can also be run
+from ipython to generate plots interactively.
+
+"""
+
+import pytest
+import numpy as np
+import scipy as sp
+import matplotlib.pyplot as plt
+import control as ct
+
+# In interactive mode, turn on ipython interactive graphics
+plt.ion()
+
+# Utility function for counting unstable poles of open loop (P in FBS)
+def _P(sys, indent='right'):
+    if indent == 'right':
+        return (sys.pole().real > 0).sum()
+    elif indent == 'left':
+        return (sys.pole().real >= 0).sum()
+    elif indent == 'none':
+        if any(sys.pole().real == 0):
+            raise ValueError("indent must be left or right for imaginary pole")
+    else:
+        raise TypeError("unknown indent value")
+
+# Utility function for counting unstable poles of closed loop (Z in FBS)
+def _Z(sys):
+    return (sys.feedback().pole().real >= 0).sum()
+
+# Decorator to close figures when done with test (to avoid matplotlib warning)
+@pytest.fixture(scope="function")
+def figure_cleanup():
+    plt.close('all')
+    yield
+    plt.close('all')
```

Comment:
Did you try to use the existing `mplcleanup` (automatically loaded from `conftest.py`)?
https://github.com/python-control/python-control/blob/d75c395083d38244fcb1ebb568c0ef61860284ad/control/tests/conftest.py#L111-L124

Now that we don't test python2 anymore, we could directly map it to [matplotlib.testing.decorators.cleanup](https://matplotlib.org/3.3.3/api/testing_api.html#matplotlib.testing.decorators.cleanup)